src/aws_encryption_sdk/keyrings/raw.py

```diff
@@ -86,11 +85,26 @@ class RawAESKeyring(Keyring):
     key_namespace = attr.ib(validator=instance_of(six.string_types))
     key_name = attr.ib(validator=instance_of(six.binary_type))
     _wrapping_key = attr.ib(repr=False, validator=instance_of(six.binary_type))
-    _wrapping_algorithm = attr.ib(repr=False, validator=instance_of(WrappingAlgorithm))
 
     def __attrs_post_init__(self):
         # type: () -> None
         """Prepares initial values not handled by attrs."""
+        key_size_to_wrapping_algorithm = {
+            wrapper.algorithm.kdf_input_len: wrapper
+            for wrapper in (
+                WrappingAlgorithm.AES_128_GCM_IV12_TAG16_NO_PADDING,
+                WrappingAlgorithm.AES_192_GCM_IV12_TAG16_NO_PADDING,
+                WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+            )
+        }
+
+        try:
+            self._wrapping_algorithm = key_size_to_wrapping_algorithm[len(self._wrapping_key)]
+        except KeyError:
+            raise ValueError(
+                "Invalid wrapping key length. Must be one of {}".format(key_size_to_wrapping_algorithm.keys())
```

Comment:
what does this error message look like, would it print out "Must be one of AES_128_GCM_IV12_TAG16_NO_PADDING, AES_192_GCM_IV12_TAG16_NO_PADDING, etc" or just the key sizes?

Reply:
It would print out the key sizes.

Reply:
I updated this to be deterministic (it already was for 3.6+ but not for 3.5 or 2.7) and updated the test to catch any changes to the set.